### DISCUSSION TITLE
First Timer - SunSynk via RS485 to USB

### DISCUSSION THREAD

**Discussion by RenierBoshoff** [slipx06/sunsynk-power-flow-card#540] [General]
So just to recap, it started with the some glitch on my side of things. With me learning and sorting through to much information.

Rather I will now post my progress below as I get along.

Setup = 8kW SunSynk > RS485 to USB converter > Raspberry Pi 3B with HA OS
Running [ KelleZA's](https://kellerza.github.io/sunsynk/) MQTT based SunSynk addon and displaying it with the SunSynk Power Flow Card.

Current stuff I need to figure out:
1) Fix UPS essential side load displaying Grid Power+Inverter Power at the moment
2) Get Shelly & Sonoff devices to function on HA an set up some sort of Smart control
3) Help 3 Customer Friends to do the same

Cheers!

_

> ### Is there an existing issue for this?
> 
> - [X] I have searched the existing issues
> 
> ### Current Behavior
> 
> In HACS, I am trying to download the sunsynk-power-flow-card
>  But I get this message:
> 
> sunsynk-power-flow-card
> Commit will be downloaded
> 
> When downloaded, this will be located in '/config/www/community/sunsynk-power-flow-card'
> **The version for this plugin can not be used with HACS.**
> 
> 
> ### Steps To Reproduce
> 
> 1 . Home Assistant Community Store
> 2 . Search for sunsynk
> 3 . Select sunsynk-power-flow-card
> 4 . Click Download
> 5 . Click download on Pop up: Commit will be downloaded
> 
> When downloaded, this will be located in '/config/www/community/sunsynk-power-flow-card'
> 
> 6. Gets error: The version for this plugin can not be used with HACS.
> ![image](https://github.com/user-attachments/assets/0af811bd-d1c9-4885-8176-c0be53062caf)
> 
> 
> ### Expected behaviour
> 
> Successful download.
> 
> ### Card Version
> 
> Raspberry Pi 3  rpi3-64
> 
> ### Home Assistant Version
> 
> Home Assistant  Core 2024.10.3 Supervisor 2024.10.3 Operating System 13.2 Frontend 20241002.3
> 
> ### Configuration
> 
> _No response_
> 
> ### Relevant log output
> 
> _No response_

_

**Comment by slipx06** [comment]
Hi. Are you running the latest version of HACS 2.0.1 ?

**Comment by RenierBoshoff** [comment]
> Hi. Are you running the latest version of HACS 2.0.1 ?

Hi!

Yes I am running the latest version.
After a reboot and some fiddling with DNS server settings I got it loaded.
![image](https://github.com/user-attachments/assets/946b61f6-5666-47e6-91b5-cfe8bbc0f67e)


Now I am at the next step, trying to get this Wave USB to RS485 Industrial Converter [https://www.robotics.org.za/W17286]
to collect the direct RS485 inverter data and display it on your Flow card...

I feel like I have no idea what I am doing even after watching a lot of videos, and reading up on the settings.

Do I need to paste this somewhere: [https://github.com/slipx06/Sunsynk-Home-Assistant-Dash/blob/main/Dashboard.yaml](url)

**Comment by RenierBoshoff** [comment]
By the way, this is the physical connections:

Inverter RS485 Port > RS485 to USB converter (the WaveShare one) > USB port of Raspberry.

**Comment by RenierBoshoff** [comment]
This afternoon the magic happened.
I think I got lost in the the multitude of options and methods.

But I finally got it working:

Next step will be to fix the Essential Load Reading which now shows the Grid side reading.
Obviously the Essential Loads on the Flow Card must match the UPS load on the Inverter flow diagram.

I will search around till I find it. Seems that there was a discussion about the 8kW SunSynk and that value not displaying the correct data.
Perhaps remap the Modbus address to the sensor.... hahaha as if I would know how to do that!

Cheers till later..

![Skermskoot 2024-10-28 163734](https://github.com/user-attachments/assets/f2b5717f-d1cb-4cb2-8b35-d91d0de6020a)

![Skermskoot 2024-10-28 165936](https://github.com/user-attachments/assets/878a4b6f-0a4e-4f32-8aca-55ace983d1f6)

**Reply by RenierBoshoff** [reply]
I have posted a video here https://github.com/kellerza/sunsynk/issues/134#issuecomment-2442536324

about the display.

I will show what I mean here as well:

Here I have my Geyser on, as a Non-Essential load.
Non Essential = 3.3kW correct.
Essential shows 3.93kW which is not what I want.

After a bit I switch the Geyser off and the "Essential" Home icon changes to a Home with a Battery (634Watt)
Why does that happen and how can I change the "Essential" value to only display the Load Side (or UPS side) of the inverter?

https://github.com/user-attachments/assets/025353c1-afd3-437c-ae95-da916acd0465

**Reply by slipx06** [reply]
Hi. Post your card config

**Reply by RenierBoshoff** [reply]
Like this?

type: custom:sunsynk-power-flow-card
cardstyle: full
show_solar: true
large_font: true
battery:
  energy: 11000
  shutdown_soc: 20
  invert_power: false
  colour: var(--energy-battery-in-color)
  show_daily: false
solar:
  colour: var(--energy-solar-color)
  show_daily: false
  mppts: 1
entities:
  use_timer_248: switch.ss_use_timer
  priority_load_243: switch.ss_priority_load
  inverter_voltage_154: sensor.ss_grid_voltage
  inverter_load_freq_192: sensor.ss_grid_frequency
  inverter_current_164: sensor.ss_inverter_current
  inverter_power_175: sensor.ss_inverter_power
  grid_connected_status_194: binary_sensor.ss_grid_connected
  inverter_status_59: sensor.ss_overall_state
  batchargeday_70: sensor.ss_day_battery_charge
  batdischargeday_71: sensor.ss_day_battery_discharge
  battery_voltage_183: sensor.ss_battery_voltage
  battery_soc_184: sensor.ss_battery_soc
  battery_power_190: sensor.ss_battery_power
  battery_current_191: sensor.ss_battery_current
  grid_power_169: sensor.ss_grid_power
  grid_buy_day_76: sensor.ss_day_grid_import
  grid_sell_day_77: sensor.ss_day_grid_export
  grid_ct_power_172: sensor.ss_grid_ct_power
  loadday_84: sensor.ss_day_load_energy
  load_frequency_192: sensor.ss_load_frequency
  essential_power: sensor.ss_essential_power
  nonessential_power: sensor.ss_non_essential_power
  aux_power_166: sensor.ss_aux_power
  solarday_108: sensor.ss_day_pv_energy
  pv1_power_186: sensor.ss_pv1_power
  pv1_v_109: sensor.ss_pv1_voltage
  pv1_i_110: sensor.ss_pv1_current

**Reply by slipx06** [reply]
That's because essential power is being calculated using the formula:

`essential = inverter_power_175 + grid_power_169 - aux_power_166 = 2kW + 1.93kW - 0kW = 3.93kW`

What does it show on the inverter screen?

The other formula for essential power uses register 167 instead of 169

`essential 1 = inverter_power_175 + grid_power_167 - aux_power_166`

**Reply by slipx06** [reply]
You can also remove these two sensors from your config and let the card calculate the values for you. Will most likely be the same as the kellarza sensors however I use 169 to calculate non_essential instead of 167

```
essential_power: sensor.ss_essential_power
nonessential_power: sensor.ss_non_essential_power
```

**Reply by RenierBoshoff** [reply]
Where can I find those calculations?


Earlier today I took this photo:
![SS8-Power Flow](https://github.com/user-attachments/assets/cc8c5fba-1b4d-402e-a0e4-9824bce15c6f)

And the HA showed this at about the same time:
![Screenshot_20241028_165413_Home Assistant](https://github.com/user-attachments/assets/87bbb84f-c9f0-4aee-b417-7cb5f4d966b2)

**Reply by slipx06** [reply]
https://github.com/kellerza/sunsynk/blob/17ce2e16395f5f836fe86328f159eb1534ce4361/src/sunsynk/definitions/single_phase.py#L94-L108

**Reply by RenierBoshoff** [reply]
> You can also remove these two sensors from your config and let the card calculate the values for you. Will most likely be the same as the kellarza sensors however I use 169 to calculate non_essential instead of 167
> 
> ```
> essential_power: sensor.ss_essential_power
> nonessential_power: sensor.ss_non_essential_power
> ```

Oh boy. So your power flow card collects its own sensor data somehow?
My apologies, I am very new to all this...
Hahaha..

So I was under the impression that I use Kellarza's addon to get the Inverter data into the MQTT which converts it to Sensors that your flow card can display?

**Reply by slipx06** [reply]
You are correct, but for those two sensors I have some internal logic that calculates their values based on other sensors you provide. 

Do you have the sensor `Load power` enabled in your addon? If so try setting that as your essential_power sensor

**Reply by RenierBoshoff** [reply]
Looks like your .py code refers to the actual Modbus register right?

**Reply by RenierBoshoff** [reply]
> You are correct, but for those two sensors I have some internal logic that calculates their values based on other sensors you provide.
> 
> Do you have the sensor `Load power` enabled in your addon? If so try setting that as your essential_power sensor

I only see these Load references in the MQTT

![image](https://github.com/user-attachments/assets/4d2982cc-91f1-4d64-9860-4f788f1a2d13)

**Reply by slipx06** [reply]
You need to add it to the addon configuration

**Reply by RenierBoshoff** [reply]
Updated this one to load_power

![image](https://github.com/user-attachments/assets/d578752d-37a6-4b79-962e-75e89c2b01ba)

**Reply by RenierBoshoff** [reply]
Mmmmm looks like it calculates the same way as the other?

Card: essential_power: sensor.ss_load_power

![image](https://github.com/user-attachments/assets/de9fd982-871c-487c-a9d4-f2003a7c676a)

**Reply by slipx06** [reply]
That value is coming from the register 178.  You can try asking for help on https://powerforum.co.za/

**Reply by RenierBoshoff** [reply]
Are we looking at the correct Modbus addresses for the Load output?

![image](https://github.com/user-attachments/assets/114908e1-9d4f-4781-9800-7efc6033acd7)

![image](https://github.com/user-attachments/assets/552df306-60f4-40ec-8c7a-73e91be45bd2)

**Reply by RenierBoshoff** [reply]
> That value is coming from the register 178. You can try asking for help on https://powerforum.co.za/

Looks like this is Load Side total power according to the Modbus register.
I will definitely look on powerforum.

Thank you for your kind assistance! Will post results...
